{"name": "Unsuccessful login user", "status": "failed", "statusDetails": {"message": "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 4s, while waiting for:\nbrowser.element(('css selector', '[data-testid=\"tab-login\"]')).click\n\nReason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[data-testid=\"tab-login\"]\"}\n  (Session info: chrome=122.0.6261.94); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nScreenshot: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.png\nPageSource: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.html", "trace": "fn = <selene.core.wait.Command object at 0x000001DAD45DDF40>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\wait.py:119: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\wait.py:61: in __call__\n    return self._fn(entity)\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\entity.py:525: in raw_click\n    element.locate().click()\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\entity.py:185: in locate\n    return self._locator()\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\locator.py:34: in __call__\n    return self._locate()\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\_browser.py:71: in <lambda>\n    Locator(f'{self}.element({by})', lambda: self.driver.find_element(*by)),\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:748: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:354: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x000001DAD45DC200>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#16 0x563fb873a683 \\\\u003Cunknown>\\\\n#17 0x563fb874b044 \\\\u003Cunknown>\\\\n#18 0x7f5729b78ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[data-testid=\"tab-login\"]\"}\nE         (Session info: chrome=122.0.6261.94); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x563fb874be93 <unknown>\nE       #1 0x563fb8443ce6 <unknown>\nE       #2 0x563fb848ee48 <unknown>\nE       #3 0x563fb848ef01 <unknown>\nE       #4 0x563fb84d23f4 <unknown>\nE       #5 0x563fb84b0edd <unknown>\nE       #6 0x563fb84cf899 <unknown>\nE       #7 0x563fb84b0c53 <unknown>\nE       #8 0x563fb8481db3 <unknown>\nE       #9 0x563fb848277e <unknown>\nE       #10 0x563fb87117cb <unknown>\nE       #11 0x563fb87157e5 <unknown>\nE       #12 0x563fb86ff0e1 <unknown>\nE       #13 0x563fb8716372 <unknown>\nE       #14 0x563fb86e31bf <unknown>\nE       #15 0x563fb873a488 <unknown>\nE       #16 0x563fb873a683 <unknown>\nE       #17 0x563fb874b044 <unknown>\nE       #18 0x7f5729b78ac3 <unknown>\n\n..\\..\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:229: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\n    @allure.epic('Login')\n    @allure.title('Unsuccessful login user')\n    @allure.link('https://www.litres.ru/', name='litres')\n    @allure.tag('web')\n    @allure.severity(Severity.MINOR)\n    @allure.label('owner', 'polushina')\n    def test_unsuccessful_login():\n        main_page.open_main_page()\n>       login_page.click_on_login()\n\ntest_login_user.py:30: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\litres_shop\\pages\\UI\\login_page.py:11: in click_on_login\n    browser.element('[data-testid=\"tab-login\"]').click()\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\entity.py:540: in click\n    self.wait.for_(\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\wait.py:150: in for_\n    return self._decorator(self)(logic)(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nfn = <selene.core.wait.Command object at 0x000001DAD45DDF40>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=getattr(reason, \"msg\", str(reason)),\n                    )\n                    # TODO: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    # TODO: should we have an option to turn on stacktrace logging?\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    # if it's a normal function, it should have __qualname__,\n                    # then use it, otherwise use str(fn)\n                    fn_name = getattr(\n                        fn,\n                        '__qualname__',\n                        getattr(fn, '__str__', lambda: str(fn))(),\n                    )\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn_name}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 4s, while waiting for:\nE                   browser.element(('css selector', '[data-testid=\"tab-login\"]')).click\nE                   \nE                   Reason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[data-testid=\"tab-login\"]\"}\nE                     (Session info: chrome=122.0.6261.94); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE                   Screenshot: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.png\nE                   PageSource: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.html\n\n..\\..\\.venv\\Lib\\site-packages\\selene\\core\\wait.py:148: TimeoutException"}, "steps": [{"name": "Открыть главную страницу \"Litres\"", "status": "passed", "start": 1719341064440, "stop": 1719341066985}, {"name": "Клик на кнопку \"Войти\"", "status": "failed", "statusDetails": {"message": "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 4s, while waiting for:\nbrowser.element(('css selector', '[data-testid=\"tab-login\"]')).click\n\nReason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[data-testid=\"tab-login\"]\"}\n  (Session info: chrome=122.0.6261.94); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nScreenshot: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.png\nPageSource: file://C:\\Users\\terry\\.selene\\screenshots\\1719340961170\\1719340961172.html\n\n", "trace": "  File \"D:\\kurs\\qaguru\\hw_progects\\qa_guru_diplom\\litres_shop\\pages\\UI\\login_page.py\", line 11, in click_on_login\n    browser.element('[data-testid=\"tab-login\"]').click()\n  File \"D:\\kurs\\qaguru\\hw_progects\\qa_guru_diplom\\.venv\\Lib\\site-packages\\selene\\core\\entity.py\", line 540, in click\n    self.wait.for_(\n  File \"D:\\kurs\\qaguru\\hw_progects\\qa_guru_diplom\\.venv\\Lib\\site-packages\\selene\\core\\wait.py\", line 150, in for_\n    return self._decorator(self)(logic)(fn)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\kurs\\qaguru\\hw_progects\\qa_guru_diplom\\.venv\\Lib\\site-packages\\selene\\core\\wait.py\", line 148, in logic\n    raise self._hook_failure(failure)\n"}, "start": 1719341066985, "stop": 1719341071392}], "start": 1719341064439, "stop": 1719341071393, "uuid": "42ee2f30-0140-4612-a8dc-c2771e5066f9", "historyId": "0a9ba86a1c936d12596d6ed0a2ee82f1", "testCaseId": "0a9ba86a1c936d12596d6ed0a2ee82f1", "fullName": "tests.UI.test_login_user#test_unsuccessful_login", "labels": [{"name": "epic", "value": "Login"}, {"name": "owner", "value": "polushina"}, {"name": "severity", "value": "minor"}, {"name": "tag", "value": "web"}, {"name": "parentSuite", "value": "tests.UI"}, {"name": "suite", "value": "test_login_user"}, {"name": "host", "value": "LAPTOP-4LM1N31D"}, {"name": "thread", "value": "14300-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.UI.test_login_user"}], "links": [{"type": "link", "url": "https://www.litres.ru/", "name": "litres"}]}